shared/controller-api/src/main/java/io/pravega/shared/controller/event/ScaleOpEvent.java

```diff
@@ -9,24 +9,40 @@
  */
 package io.pravega.shared.controller.event;
 
-import lombok.AllArgsConstructor;
 import lombok.Data;
 
 import java.util.AbstractMap;
 import java.util.List;
 import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
 
 @Data
-@AllArgsConstructor
 public class ScaleOpEvent implements ControllerEvent {
     private static final long serialVersionUID = 1L;
     private final String scope;
     private final String stream;
     private final List<Integer> segmentsToSeal;
-    private final List<AbstractMap.SimpleEntry<Double, Double>> newRanges;
+    private final List<AbstractMap.SimpleEntry<Long, Long>> newRanges;
```

Comment:
`Map.Entry`?

Reply:
That will mean changing signatures of methods where these values are used which expects it to be AbstractMap.SimpleEntry.